client/landing/gutenboarding/stores/onboard/actions.ts

```diff
@@ -6,53 +6,65 @@ import { VerticalsTemplates } from '@automattic/data-stores';
 /**
  * Internal dependencies
  */
-import { ActionType, SiteVertical } from './types';
+import { SiteVertical } from './types';
 
 type Template = VerticalsTemplates.Template;
 
 export const setDomain = (
 	domain: import('@automattic/data-stores').DomainSuggestions.DomainSuggestion | undefined
 ) => ( {
-	type: ActionType.SET_DOMAIN as const,
+	type: 'SET_DOMAIN' as const,
```

Comment:
This is just a question for my own curiosity (and wrapping my head around TS features), what's the value of adding `as const` on a string literal? Does this prevent the `type` key from being changed?

Reply:
Without it the `type` prop would still be a `string`, not a string literal unfortunately.

Another interesting way of doing it is:
```typescript
return <const>{
	type: 'SET_DOMAIN',
	domain,
};
```

Which not only makes typeof `type` the correct string literal, but it also marks all the props as `readonly`.

Hover your mouse over the types in [this playground example](http://www.typescriptlang.org/play/?ssl=14&ssc=1&pln=15&pc=1#code/MYewdgzgLgBAhsKBLcBBGBeGBvAUDAmKATwAcBTALhgHJUBhAFQEkB5AOQH1Ub4IZQkKLgC+Ably4A9FJiMAFuSJklIAGZF5SftpgADOkzZcee3AmRoAdCQoTJkwdHiIUYAEKYc+Qraq0GFg5OdxpRexk5RWUKGHVNXV09aAAnJDAAczMLN3cbFXsHJ1gc8HovAB5igD48Qhj-QyCuejDxSUiFJT84jSgtHX4DQONOVuzXMvy7XCA) to see it in action.

Reply:
Ah, gotcha! Thanks for example, that makes it clear to me now :)

Reply:
I'd like the reducers to all `return { … } as const` if anyone wants to take that task on 👍